module_prototyper/models/module_prototyper.py

```diff
@@ -433,14 +478,30 @@ def generate_data_files(self):
             target.setdefault(model, model_obj.browse([]))
             target[model] |= model_obj.search(safe_eval(ir_filter.domain))
 
+
         res = []
         for prefix, model_data, file_list in [
                 ('data', data, self._data_files),
                 ('demo', demo, self._demo_files)]:
+
+            items = []
             for model_name, records in model_data.iteritems():
+                dependencies = []
+                for f in records._fields:
+                    if records._fields[f].comodel_name != None:
+                        dependencies.append(records._fields[f].comodel_name)
+                # ensure unique values of comodel_types
+            items.append( [model_name, dependencies, records] )
```

Comment:
If I have no data - 
server-tools/module_prototyper/models/module_prototyper.py", line 494, in generate_data_files
items.append( [model_name, dependencies, records] )
local variable 'model_name' referenced before assignment